dali/kernels/imgproc/resample/resampling_batch.cu

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime.h>
+#include "dali/kernels/imgproc/resample/resampling_batch.h"
+#include "dali/kernels/imgproc/resample/bilinear_impl.cuh"
+#include "dali/kernels/imgproc/resample/nearest_impl.cuh"
+#include "dali/kernels/imgproc/resample/resampling_impl.cuh"
+
+namespace dali {
+namespace kernels {
+
+template <int which_pass, typename Output, typename Input>
+__global__ void BatchedSeparableResampleKernel(Output *out, const Input *in,
+    const SeparableResamplingSetup::SampleDesc *samples, const SampleBlockInfo *block2sample) {
+  // find which part of which sample this block will process
+  SampleBlockInfo sbi = block2sample[blockIdx.x];
+  SeparableResamplingSetup::SampleDesc sample = samples[sbi.sample];
+  int in_stride, out_stride;
+  Output *sample_out;
+  const Input *sample_in;
+  int blocks;
+
+  DeviceArray<int, 2> in_shape, out_shape;
+
+  in_stride = sample.strides[which_pass];
+  out_stride = sample.strides[which_pass+1];
+  sample_in = in + sample.offsets[which_pass];
+  sample_out = out + sample.offsets[which_pass+1];
+  blocks = sample.block_count.pass[which_pass];
+  in_shape = sample.shapes[which_pass];
+  out_shape = sample.shapes[which_pass+1];
+
+  int block = sbi.block_in_sample;
+
+  // 0 = vertical axis, 1 = horizontal axis
```

Comment:
Add more info about why we have this bit magic bellow.

Reply:
Would `(which_pass == 0) != (sample.order == SeparableResamplingSetup::VertHorz)` be more readable in your opinion?

Reply:
It is still not clear why you are doing this.

Reply:
I'm not questioning code but if user will know why is that happening.

Reply:
As below, `0` and `1` are not that descriptive as passes, maybe some names?

Reply:
@awolant: `which_pass` is just ordinal of the pass - now there are two, for 3D resampling there would be three. As for axis - that's a bit more complicated, but for HCW layout that's the ordinal of the dimension (H -> 0, W -> 1). For 3D data it would change meaning (D -> 0, H -> 1, W -> 2), so naming them isn't future-proof.

Reply:
@JanuszL: Sure, it's a bit of a trickery for the 2D case. I can add a comment there.

Reply:
> so naming them isn't future-proof
Right

Reply:
Magic removed.